### PR TITLE
Forward-compatible HTTPS options

### PR DESCRIPTION
**Pull request by DullReferenceException**
I recently discovered that `create-servers` (used by `slay`) was not preserving my `requestCert` option. By changing it to pass through all https options by default, it will now enable client certs and other SSL features which are currently blocked (most likely unintentionally). Also changed the code so it does not mutate the original options.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -122,54 +122,54 @@ module.exports = function createServers(options, listening) {
...
       // Load default SSL key, cert and ca(s).
       //
-      key:  fs.readFileSync(path.resolve(ssl.root, ssl.key)),
-      cert: fs.readFileSync(path.resolve(ssl.root, ssl.cert)),
-      ca:   ssl.ca && ssl.ca.map(
-        function (file) {
-          return fs.readFileSync(path.resolve(ssl.root, file));
-        }
+      key: key,
+      cert: cert,
```
Comment (indexzero) [inline index.js]: This seems to be orthogonal to the change, no?

File: index.js [hunk 2]
```diff
@@ -122,54 +122,54 @@ module.exports = function createServers(options, listening) {
...
 
-    if (ssl.ca && !Array.isArray(ssl.ca)) {
-      ssl.ca = [ssl.ca];
+    if (ca && !Array.isArray(ca)) {
+      ca = [ca];
     }
 
-    log('https | listening on %d', port);
-    server = https.createServer({
+    var finalHttpsOptions = Object.assign({}, ssl, {
```
Comment (indexzero) [inline index.js]: Please use `require('object-assign')` so that we are backwards compatible. My litmus test here is making best efforts to support the current "Maintenance mode" versions of Node.js until they expire.

File: index.js (hunk 1 above)
Comment (DullReferenceException) [inline index.js]: How so? It should have the same value as before; it's just initialized above and doesn't mutate the original options that were passed in.

File: index.js (hunk 2 above)
Comment (DullReferenceException) [inline index.js]: Done; could you re-review @indexzero?

File: index.js (hunk 1 above)
Comment (indexzero) [inline index.js]: Sure, but why move it at all?

Comment (indexzero) [inline index.js]: In other words: the change is not semantic. Or am I missing something?

Comment (DullReferenceException) [inline index.js]: Sure, it doesn't have to be moved in this case; it's just code ordering so you have this sequence:

1. Extract data from original options
2. Transform data if needed
3. Build new options from that data

Comment (DullReferenceException) [inline index.js]: Personally I find it easier to follow, but if it's important to you, I can inline more stuff instead.

Comment (DullReferenceException) [inline index.js]: Done, @indexzero

**Comment by indexzero** [comment]
Thanks for your contribution, this will be a minor release because of the new feature.
